libraries/AP_HAL_ChibiOS/sdcard.cpp

```diff
@@ -81,7 +81,9 @@ bool sdcard_init()
         printf("Successfully mounted SDCard (slowdown=%u)\n", (unsigned)sd_slowdown);
 
         // Create APM Directory if needed
-        AP::FS().mkdir("/APM");
+        if (AP::FS().mkdir("/APM") != 0 && errno != EEXIST) {
```

Comment:
we are much better off handling this error at the logging level. Returning false will cause it to continually try and mount then fail to mkdir the directory for the whole flight. It is just wasting time.
If the mount didn't fail then we should return true. The attempt to mkdir is just to save having to do that in each subsystem that tries to use the sd. We don't want to check the result